File: Connection/RequestPacket.cs
using System.Net.Sockets;

namespace LANchat.Connection
{
    public class RequestPacket : AbstractTcpPacket
    {
        public RequestPacket() : base(PacketDataType.REQUEST) { }

        public override void Apply(MainWindow mw, ChatClient client)
        {
            client.Client.Send(new InformationPacket(App.Settings).ToByteArray());
        }
    }
}

File: connection/requestpacket.cpp
#include "requestpacket.h"
#include "informationpacket.h"
#include "client.h"

extern AppSettings SETTINGS;

namespace Connection
{
    RequestPacket::RequestPacket() : AbstractTcpPacket(PacketDataType::REQUEST) { }

    void RequestPacket::Apply(MainWindow& mw, ChatClient& client)
    {
        InformationPacket information(SETTINGS);
        client.SendPacket(information);
    }
}// namespace Connection
